File: 2015/pratica/kfree/att/kfree.cpp
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int trova(int N, int K, int *insieme) {
    // Mettete qui il codice della soluzione
    return 42;
}


int insieme[MAXN+1];

int main(){
    FILE *fr, *fw;
    int N, K, i;
    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");

    assert(2 == fscanf(fr, "%d %d", &N, &K));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", insieme+i));

    fprintf(fw, "%d\n", trova(N, K, insieme));

    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed1/pratica/kfree/att/kfree.pas
function trova(N, K: longint; var insieme: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    trova := 42;
end;


const
    MAXN = 100000;
var
    insieme : array[0..MAXN-1] of longint;
    N, K, i : longint;
    fr, fw  : text;
begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, K);
    for i:=0 to N-1 do
        readln(fr, insieme[i]);

    writeln(fw, trova(N, K, insieme));
    close(fr);
    close(fw);
end.
